build.gradle.kts

```diff
@@ -1,8 +1,33 @@
 plugins {
   id("otel.java-conventions")
+  id("java-library")
+  id("maven-publish")
```

Comment:
I think we should use the otel plugin, like [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/build.gradle.kts#L13)

Reply:
The otel plugin isn't published. The source code lives in the repository in `buildsrc` [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts).

Still, we should similarly organize the project with convention plugins.

Reply:
@jack-berg are you suggesting writing similar plugin as [this](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts) in this repo as well?
If yes, can you please tell me advantage of using it over standard maven publish pugin?